my-vault/internal/storage/data_access.go

```diff
@@ -26,6 +26,10 @@ func NewInMemorySecretRepo() *InMemorySecretRepo {
 // so we construct the secret object in the handler and we pass it to this function to actually add to our repository
 // returns nothing if all goes well, and also now I have a localised mutex instead of global one (funny if I do repo.mu.Unlock its giving an error)
 func (repo *InMemorySecretRepo) PutNewSecret(secret domain.Secret) error{
+	if repo == nil {
+		return errors.New("repository is unavailable")
+	}
```

Comment:
This is an interesting error check, because as you probably noticed it stops the program from panicing when called on a nil repo. However, often it can be desirable for the program to "fail noisily" in fundamental code issues like a nil repo. The repo should never really be nil if the code is written correctly, and handling it like this can actually hide the problem. That said, it's up to you if you want to keep this. I think if you do want to keep it, we should add it to the other methods on the repo as well.